cmd/tendermint/commands/rollback.go

```diff
@@ -0,0 +1,46 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/internal/state"
+)
+
+var RollbackStateCmd = &cobra.Command{
+	Use:   "rollback",
+	Short: "rollback tendermint state by one height",
+	Long: `
+Rollbacking state should be performed in the event of an incorrect applicate state transition
+whereby Tendermint has persisted an incorrect app hash and is thus unable to make 
+progress. This takes a state at height n and overwrites it with the state at height n - 1. 
+It is expected that the application also rolls back to height n - 1. No blocks are removed 
+so upon restarting Tendermint, the transactions in block n will be reexecuted against the 
+application.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		height, hash, err := RollbackState(config)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Rolled back state to height %d and hash %v", height, hash)
+		return nil
+	},
+}
+
+// RollbackState takes a state at height n and overwrites it with the state
+// at height n - 1. Note state here refers to tendermint state not application state.
+// Returns the latest state height and app hash alongside an error if there was one.
+func RollbackState(config *cfg.Config) (int64, []byte, error) {
```

Comment:
Does this need to be exported? I think library consumers will use the function in the `state` package directly, right?

Reply:
The state package is internal however. I was unsure which package would be the most logical to expose this function. I didn't want to make an entirely new package so I put it here

Reply:
Oh, right, I see what you mean. Hmm. It seemed weird to have other packages importing parts of the command-line tool. Might it make sense to create a separate "cli-support" type package that is public, and imported by the CLI so that we don't have to worry about breaking things when we modify the CLI?

Reply:
We have a package `libs/cli`. Do you think this would be a better place to have the function?

Reply:
Yes, I think it might—we documented `libs` packages as "importable packages", and I think that would be a better precedent. I think it shouldn't be too painful a change?

Reply:
nope I'll shift it across